### PR TITLE
Calendar list display

### PR DESCRIPTION
**Pull request by Akhill2020**
**Description:** Updated code to display calendar list in admin.

### CODE REVIEW

**Comment by rosinghal** [comment]
@Akhill2020 please change base branch

File: includes/admin/auth-service-actions.php [hunk 1]
```diff
@@ -45,6 +45,7 @@ public function __construct()
 		$post_type = get_post_type();
 		if($post_type = 'calendar'){
 			add_action('admin_init', [$this, 'oauth_check_iftoken_expired']);
+			add_action('admin_init', [$this, 'auth_get_calendarlist']);
```
Comment (rosinghal) [inline includes/admin/auth-service-actions.php:48]: This should not be needed here, but where we need to display list

File: includes/admin/auth-service-actions.php [hunk 2]
```diff
@@ -120,6 +121,28 @@ public function oauth_check_iftoken_expired(){
 		}
 
 	}
+
+	/*
+	* Get calendar list
```
Comment (rosinghal) [inline includes/admin/auth-service-actions.php:126]: Add version number here and return type

**Review by rosinghal — CHANGES_REQUESTED** [review]

**Comment by Akhill2020** [comment]
> @Akhill2020 please change base branch

Done

File: includes/admin/auth-service-actions.php (hunk 1 above)
Comment (Akhill2020) [reply includes/admin/auth-service-actions.php]: removed

File: includes/admin/auth-service-actions.php [hunk 3]
```diff
@@ -40,10 +36,16 @@ public function __construct()
...
+
+		$post_type = '';
+
+		if (isset($_GET['post_type']) && 'calendar' === $_GET['post_type']) {
+			$post_type = esc_attr($_GET['post_type']);
+		} elseif (isset($_GET['post']) && !empty($_GET['post'])) {
+			$post_id = tesc_attr($_GET['post']);
+			$post_type = get_post_type($post_id);
+		}
+		if (isset($post_type) && !empty($post_type) && 'calendar' === $post_type) {
```
Comment (rosinghal) [inline includes/admin/auth-service-actions.php:48]:
```suggestion
		if ('calendar' === $post_type) {
```
This should be enough, right?

File: includes/admin/auth-service-actions.php [hunk 4]
```diff
@@ -60,69 +62,88 @@ public function oauth_deauthenticate_site()
...
+	}
+
+	/*
+	 * Get calendar list
+	 */
+	public function auth_get_calendarlist()
+	{
+		$send_data = [
+			'site_url' => self::$my_site_url,
+			'auth_code' => '',
```
Comment (rosinghal) [inline includes/admin/auth-service-actions.php:133]: auth code needs to be passed

File: includes/admin/auth-service-actions.php [hunk 5]
```diff
@@ -60,69 +62,88 @@ public function oauth_deauthenticate_site()
...
+		];
+		$request = wp_remote_post(self::$url . 'auth_get_calendarlist', [
+			'method' => 'POST',
+			'body' => $send_data,
+			'cookies' => [],
+		]);
+
+		$response = wp_remote_retrieve_body($request);
```
Comment (rosinghal) [inline includes/admin/auth-service-actions.php:141]: What will happen if response is unsuccessful?

File: includes/admin/auth-service-actions.php (hunk 3 above)
Comment (Akhill2020) [reply includes/admin/auth-service-actions.php]: Update code

File: includes/admin/auth-service-actions.php (hunk 4 above)
Comment (Akhill2020) [reply includes/admin/auth-service-actions.php]: Added auth code

File: includes/admin/auth-service-actions.php (hunk 5 above)
Comment (Akhill2020) [reply includes/admin/auth-service-actions.php:147]: Code updated

**Review by rosinghal — APPROVED** [review]

**Comment by rosinghal** [comment]
@Akhill2020 can this PR be closed now?

**Pull request closed without merging**